common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleIntegrationSuite.java

```diff
@@ -168,7 +180,9 @@ private FetchResult fetchBlocks(
       String execId,
       String[] blockIds,
       TransportConf clientConf,
-      int port) throws Exception {
+      int port,
+      int shuffleGenerationId,
```

Comment:
maybe just an `Integer shuffleGenerationId`, and use null to indicate fetching data block.